spec/common/providers/identity.spec.ts

```diff
@@ -85,4 +90,857 @@ describe('identity', () => {
+      );
+    });
+
+    it('should return the correct public key', () => {
+      expect(
+        identity.getPublicKey(
+          {
+            alg: identity.JWT_ALG,
+            kid: '123456',
+          },
+          {
+            '123456': '7890',
+            '2468': '1357',
+          }
+        )
+      ).to.eq('7890');
+    });
+  });
+
+  describe('isAuthorizedCloudFunctionURL', () => {
+    it('should return false on a bad gcf direction', () => {
+      expect(
+        identity.isAuthorizedCloudFunctionURL(
+          `https://us-central1-europe-${PROJECT}.cloudfunctions.net/function-1`,
+          PROJECT
+        )
+      ).to.be.false;
+    });
+
+    it('should return false on a bad project', () => {
+      expect(
+        identity.isAuthorizedCloudFunctionURL(
+          `https://us-central1-${PROJECT}-old.cloudfunctions.net/function-1`,
+          PROJECT
+        )
+      ).to.be.false;
+    });
+
+    it('should return true on a good url', () => {
+      expect(identity.isAuthorizedCloudFunctionURL(VALID_URL, PROJECT)).to.be
+        .true;
+    });
+  });
+
+  describe('checkDecodedToken', () => {
+    it('should throw on unauthorized function url', () => {
+      expect(() =>
+        identity.checkDecodedToken(
+          {
+            aud: `fake-region-${PROJECT}.cloudfunctions.net/fn1`,
+          } as identity.DecodedJwt,
+          PROJECT
+        )
+      ).to.throw('Provided JWT has incorrect "aud" (audience) claim.');
+    });
+
+    it('should throw on a bad iss property', () => {
+      expect(() =>
+        identity.checkDecodedToken(
+          {
+            aud: VALID_URL,
+            iss: `https://someissuer.com/a-project`,
+          } as identity.DecodedJwt,
+          PROJECT
+        )
+      ).to.throw(
+        `Provided JWT has incorrect "iss" (issuer) claim. Expected "${identity.JWT_ISSUER}${PROJECT}" but got "https://someissuer.com/a-project".`
+      );
+    });
+
+    it('should throw if sub is not a string', () => {
+      expect(() =>
+        identity.checkDecodedToken(
+          ({
+            aud: VALID_URL,
+            iss: `${identity.JWT_ISSUER}${PROJECT}`,
+            sub: {
+              key: 'val',
+            },
+          } as unknown) as identity.DecodedJwt,
+          PROJECT
+        )
+      ).to.throw('Provided JWT has no "sub" (subject) claim.');
+    });
+
+    it('should throw if sub is empty', () => {
+      expect(() =>
+        identity.checkDecodedToken(
+          {
+            aud: VALID_URL,
+            iss: `${identity.JWT_ISSUER}${PROJECT}`,
+            sub: '',
+          } as identity.DecodedJwt,
+          PROJECT
+        )
+      ).to.throw('Provided JWT has no "sub" (subject) claim.');
+    });
+
+    it('should throw if sub length is larger than 128 chars', () => {
+      const str = [];
+      for (let i = 0; i < 129; i++) {
+        str.push(i);
```

Comment:
nit: this is going to be way longer than 129 chars. I'd probably do something like `sub: "a".repeat(129)`

Reply:
Fixed